src/Sulu/Bundle/RouteBundle/Resources/js/index.js

```diff
@@ -12,3 +13,15 @@ fieldRegistry.add(
         options: {history: true},
     }
 );
+
+fieldRegistry.add(
+    'page_tree_route',
+    PageTreeRoute,
+    {
+        historyResourceKey: 'routes',
+        modeResolver: () => {
+            return Promise.resolve('leaf');
+        },
+        options: {history: true},
```

Comment:
We should probably think about which of these options make sense in that case. I think we could remove all of them, unless you are (or intent to) register these component with other options as well.

Reply:
WDYT about this?

Reply:
I don't know about future plans with the PageTreeRoute. I don't need these options. Just added them, because i saw them in ResourceLocator and from the point of extensibility I expected the same behavior here.

Reply:
These are my standpoints on each one of them:

- Since the `page_tree_route` always creates some `Route` object from the `RouteBundle`, I would argue that the `historyResourceKey` will never change, so that could be hardcoded.
- I don't know if the mode can be changed, depending on that it might also be possible to remove the `modeResolver`. I am sure @wachterjohannes can tell us more about it.
- The `options` are passed to the `ResourceLocatorHistory` (could probably have been named a little bit better...) and AFAIK the `routes` resource always needs this `history` flag. So this could also be hardcoded and not be made configurable.